docs/firestore/crud-tutorial

```diff
@@ -0,0 +1,792 @@
+    AngularFirestoreModule
+  ],
+  declarations: [ AppComponent ],
+  bootstrap: [ AppComponent ]
+})
+export class AppModule {}
+```
+
+### 6. Inject `AngularFirestore`
+
+Open `/src/app/app.component.ts` and import `AngularFirestore`.
+
+```ts
+import { Component } from '@angular/core';
+import { AngularFirestore } from '@angular/fire/compat/firestore';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+
+  constructor(firestore: AngularFirestore)
+}
+
+```
+
+### 7. Set up Firestore collection
+
+Now open your Firebase Console and create your Firestore database, in test mode. Make a `collection` and call it `greatest-computer-scientists`. Firestore will ask you to make one document. Call the document `Charles Babbage` and put in two fields, both strings:
+
+```
+name: Charles Babbage
+accomplishment: Built first computer
+```
+
+### 8. Bind an Observable to the Firestore collection
+
+Import `Observable` from `rxjs`. Then make an instantiation of the `Observable` class.
+
+In `/src/app/app.component.ts`:
+
+```ts
+import { Component } from '@angular/core';
+import { AngularFirestore } from '@angular/fire/compat/firestore';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css']
+})
+export class AppComponent {
+  scientists: Observable<any[]>;
+  constructor(private firestore: AngularFirestore) {
+    this.scientists = firestore.collection('greatest-computer-scientists').valueChanges();
+  }
+}
+```
+
+The constructor makes an instantiation of the `AngularFirestore` class and we call it `firestore` for convenience and clarity. You must specify this as `public`, `protected`, or `private`. It doesn't matter which, they all work. It won't work if you don't specify one of these:
+
+```ts
+  constructor(firestore: AngularFirestore) { // doesn't work
+}
+```
+
+I know what you're thinking, the [TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/2/classes.html#public) says
+
+```
+ Because public is already the default visibility modifier, you don’t ever need to write it on a class member.
+```
+
+That's not right, or not right in a way that doesn't require a lot of explaining. Here's a discussion of why we have to [use a visibility modifiers here](https://github.com/angular/angularfire/issues/3173).
+
+Also in the constructor we link to our database. Now we can call our database and store the results in use `this.scientists`.
+
+#### Making an Interface or Type for a Firebase Collection
+
+Did you notice that we set a type to `any`:
+
+```js
+scientists: Observable<any[]>;
+```
+
+TypeScript hates it when we declare `any`. Surely we can declare an `interface` or `type`:
+
+```js
+export declare type Scientists = {
+  accomplishment: string;
+  name: string;
+}
+
+export class HomePageComponent {
+  scientists: Observable<Scientists>;
```

Comment:
In your example Scientists isn't an array. imo the type should also be "Scientist" instead of "Scientists", or make "Scientists" be: 

```js
export type Scientists = Scientist[];
```

Reply:
> In your example Scientists isn't an array. imo the type should also be "Scientist" instead of "Scientists", or make "Scientists" be:
> 
> ```js
> export type Scientists = Scientist[];
> ```

Good catch! That was old code that I didn't update. I want to show the complete code but this makes it hard to catch mistakes as the code changes. The IntelliJ EduTools plugin solves this problem.

Reply:
The Firestore Data Converter interface looks like what I need to make types for Firestore queries. I'll study this and update my tutorial. Thanks!